### PR TITLE
MAINTAINERS.rst: cleanups, clarifications

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
Just some cleanups and clarifications on how I'd like to do this going
forward. The main clarifications are around SCHEMA_VERSION, since this
is changing in 0.8 and I want to make sure we're all on the same page.

### CODE REVIEW

**Comment by mbolivar-nordic** [comment]
(I've gone through these steps locally in a v0.8-branch I have yet to push, but I want to get some review on this before moving forward.)

File: MAINTAINERS.rst [hunk 1]
```diff
@@ -1,8 +1,14 @@
-Notes for maintainers cutting a release.
+Notes for maintainers.
 
 Pre-release test plan
 ---------------------
 
+0. If no release branch exists, fork the version numbers in the release branch
+   (vX.Y-branch) and the master branch. See "Cutting a release branch", below,
```
Comment (tejlmand) [inline MAINTAINERS.rst:7]: >  fork the version numbers in the release branch (vX.Y-branch) and the master branch. 

I'm not sure exactly what you mean by this. I cannot fork from two branches.
Did you means something like:
> fork the version number from the master branch into a new release branch (vX.Y-branch)
   (vX.Y-branch).

File: MAINTAINERS.rst [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
+   for details.
+
+   The rest of these steps should be done in the release branch.
```
Comment (tejlmand) [inline MAINTAINERS.rst:10]: I know what you mean, but should we make it even clearer ?
Switch to the newly created release branch:
```
git checkout vX.Y-branch`
```

File: MAINTAINERS.rst [hunk 3]
```diff
@@ -19,14 +25,8 @@ Pre-release test plan
...
 
 3. Build alpha N (N=1 to start, then N=2 if you need more commits, etc.) and
-   upload to pypi.
-
-   Make sure to check if west.manifest.SCHEMA_VERSION also needs an update
-   before uploading. ::
-
-     git clean -ffdx
-     WEST_VERSION=X.YaN python3 setup.py sdist bdist_wheel
-     twine upload -u zephyr-project dist/*
```
Comment (tejlmand) [inline MAINTAINERS.rst:29]: Why are the example commands removed ?

**Review by tejlmand — CHANGES_REQUESTED** [review]
Nice improvement.
Just one sentence where i'm not clear what you actually mean.

And an additional suggestion.
For all places where branches, commits, etc. are used, then it would be nice to have the `git <command>` available, as it makes it easier to just copy and adjust the command.
Also, if someone else joins, they can see what commands was actually used in each step.

File: MAINTAINERS.rst (hunk 1 above)
Comment (mbolivar-nordic) [reply MAINTAINERS.rst:7]: The two branches fork from each other.

Precondition: `vX.Y-branch` does not exist, `master` is at version `X.(Y-1)`.
Postcondition: `v.X.Y-branch` exists and is at version `vX.Y.0a1`, `master` is at version `vX.Y.99`.

File: MAINTAINERS.rst (hunk 3 above)
Comment (mbolivar-nordic) [reply MAINTAINERS.rst:29]: >  Why are the example commands removed ?

They weren't. There are two copies. I eliminated the redundant copy, leaving the single point of truth in the details section.

**Comment by mbolivar-nordic** [comment]
@tejlmand thanks for review, now addressed.

File: MAINTAINERS.rst (hunk 1 above)
Comment (tejlmand) [reply MAINTAINERS.rst:7]: much clearer, could you write that instead ?

Comment (mbolivar-nordic) [reply MAINTAINERS.rst:7]: >  could you write that instead ?

Sure. Added to the details section.

**Review by tejlmand — APPROVED** [review]

**Pull request merged**